src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -4494,5 +4491,167 @@ public override ((int b, int a), (int b, int a)) M(((int a, int b), (int b, int
+" + trivial2uple;
+
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular.WithTuplesFeature().WithPatternsFeature());
+            comp.VerifyDiagnostics(
+                // (6,19): error CS1525: Invalid expression term 'int'
+                //         if (o is (int, int) t)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "int").WithArguments("int").WithLocation(6, 19),
+                // (6,24): error CS1525: Invalid expression term 'int'
+                //         if (o is (int, int) t)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "int").WithArguments("int").WithLocation(6, 24),
+                // (6,29): error CS1026: ) expected
+                //         if (o is (int, int) t)
+                Diagnostic(ErrorCode.ERR_CloseParenExpected, "t").WithLocation(6, 29),
+                // (6,30): error CS1002: ; expected
+                //         if (o is (int, int) t)
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, ")").WithLocation(6, 30),
+                // (6,30): error CS1513: } expected
+                //         if (o is (int, int) t)
+                Diagnostic(ErrorCode.ERR_RbraceExpected, ")").WithLocation(6, 30),
+                // (6,29): error CS0103: The name 't' does not exist in the current context
+                //         if (o is (int, int) t)
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "t").WithArguments("t").WithLocation(6, 29)
+                );
+        }
+
+        [Fact]
+        public void TupleTypeWithPatternIs2()
+        {
+            var source = @"
+class C
+{
+    void Match(object o)
+    {
+        if (o is (int a, int b)) { }
+    }
+}
+" + trivial2uple;
+
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular.WithTuplesFeature().WithPatternsFeature());
+            comp.VerifyDiagnostics(
+                // (6,32): error CS1003: Syntax error, '=>' expected
+                //         if (o is (int a, int b)) { }
+                Diagnostic(ErrorCode.ERR_SyntaxError, ")").WithArguments("=>", ")").WithLocation(6, 32),
+                // (6,32): error CS1525: Invalid expression term ')'
+                //         if (o is (int a, int b)) { }
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, ")").WithArguments(")").WithLocation(6, 32)
+                );
+        }
+
+        [Fact]
+        public void TupleDeclarationWithPatternIs()
+        {
+            var source = @"
+class C
+{
+    void Match(object o)
+    {
+        if (o is (1, 2)) { }
+    }
+}
+" + trivial2uple;
+
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular.WithTuplesFeature().WithPatternsFeature());
+            comp.VerifyDiagnostics(
+                // (6,18): error CS0150: A constant value is expected
+                //         if (o is (1, 2))
+                Diagnostic(ErrorCode.ERR_ConstantExpected, "(1, 2)").WithLocation(6, 18)
```

Comment:
This one is a little worrying. It suggests that `(1, 2)` is being parsed as a tuple expression, which should not be possible in this context.